src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -556,14 +557,16 @@ def list_apps(client, app_id=None, display_name=None, identifier_uri=None, query
     return client.list(filter=(' and '.join(sub_filters)))
 
 
-def list_sps(client, spn=None, display_name=None, query_filter=None):
+def list_sps(client, spn=None, display_name=None, query_filter=None, appid=None):
```

Comment:
please check out my comments below. If you use `_resolve_service_principal`, then this change can be reverted

Reply:
Reverted